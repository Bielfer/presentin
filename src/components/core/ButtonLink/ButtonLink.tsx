import clsx from 'clsx';
import Link from 'next/link';
import { FC } from 'react';
import { Size, sizes, variantStyles } from '@/components/core/Button';
import { Props } from '@/components/core/MyLink';

interface ButtonProps extends Props {
  size?: Size;
  iconLeft?: React.ComponentType<React.ComponentProps<'svg'>>;
  iconRight?: React.ComponentType<React.ComponentProps<'svg'>>;
}

const ButtonLink: FC<ButtonProps> = ({
  className,
  children,
  variant = 'white',
  href,
  size = 'md',
  iconLeft: IconLeft,
  iconRight: IconRight,
  ...props
}) => (
  <Link href={href || ''} passHref>
    <a
      className={clsx(
        'inline-flex items-center font-medium rounded-lg shadow-sm',
        variantStyles[variant],
        sizes[size],
        className
      )}
      {...props}
    >
      {IconLeft && <IconLeft className="-ml-0.5 mr-2 h-5 w-5 flex-shrink-0" />}
      {children}
      {IconRight && (
        <IconRight className="ml-2 -mr-0.5 h-5 w-5 flex-shrink-0" />
      )}
    </a>
  </Link>
);

export default ButtonLink;
