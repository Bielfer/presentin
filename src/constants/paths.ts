const paths = {
  home: '/',
  pricing: '/#pricing',
  features: '/#features',
  createPresentin: '/presentin/new',
  login: '/login',
  presentinMessage: (presentinId: string) => `${presentinId}/message`,
};

export default paths;
