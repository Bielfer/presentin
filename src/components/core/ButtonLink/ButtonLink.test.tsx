import paths from '@/constants/paths';
import { render, screen } from '@testing-library/react';
import ButtonLink from '.';

describe('ButtonLink', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <ButtonLink href={paths.home}>Test</ButtonLink>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a href', () => {
    const href = '/test';
    render(<ButtonLink href={href}>Test</ButtonLink>);

    const RenderedButtonLink = screen.getByRole('link');

    expect(RenderedButtonLink).toHaveAttribute('href', href);
  });
});
