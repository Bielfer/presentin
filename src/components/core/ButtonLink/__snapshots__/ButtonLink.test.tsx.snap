// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="inline-flex items-center font-medium rounded-lg shadow-sm border border-gray-300 text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 px-4 py-2 text-sm"
    href="/"
  >
    Test
  </a>
</DocumentFragment>
`;
